### PR TITLE
MAINT: Adapt to R107 CSS button color changes

### PR DESCRIPTION
**Pull request by bananarama92**
Note: this PR should not be merged until R107 is live and R106 support is ready to be dropped.

This PR adapts to the CSS button changes introduced in [BondageProjects/Bondage-College#5127](https://gitgud.io/BondageProjects/Bondage-College/-/merge_requests/5127?commit_id=d265b63cea043309cc7cd14cd1ccb2eab665feaa), which among others splits the managing of button background colors into three distinct variables: `--button-color`, `--disabled-color` and `--hover-color`; variables that, in most cases, should modified instead of directly setting `background-color` itself. 

The goal of these changes was to make it easier to only override the "default" background color while leaving the styling of disabled and hover colors alone. As such, the WCE buttons should now display the expected hovering color (cyan) again upon, well..., hovering over them, as the hovering color was previously overriding by the id-level `background-color` declaration of the `ignore` and `preserve` states.

### CODE REVIEW

File: src/functions/wceStyles.ts [hunk 1]
```diff
@@ -367,11 +367,11 @@ export default function wceStyles(): void {
   }
   #wce-chat-baby-talk[data-state="ignore"]::before,
   #wce-chat-stutters[data-state="ignore"]::before {
-    background-color: #bbb;
+    --button-color: #bbb;
```
Comment (KittenApps) [inline src/functions/wceStyles.ts:370]: This is inside a template literal, so you could so something like:

```suggestion
    ${GameVersion === "R104" ? "background" : "--button"}-color: #bbb;
```

File: src/functions/wceStyles.ts [hunk 2]
```diff
@@ -367,11 +367,11 @@ export default function wceStyles(): void {
...
   }
   #wce-chat-baby-talk[data-state="preserve"]::before,
   #wce-chat-stutters[data-state="preserve"]::before {
-    background-color: #666;
+    --button-color: #666;
```
Comment (KittenApps) [inline src/functions/wceStyles.ts:374]: same

```suggestion
    ${GameVersion === "R104" ? "background" : "--button"}-color: #666;
```

**Review by KittenApps — CHANGES_REQUESTED** [review]
We could make it compatible with both R106 and R107.

File: src/functions/wceStyles.ts (hunk 2 above)
Comment (bananarama92) [reply src/functions/wceStyles.ts]: Applied as of https://github.com/KittenApps/WCE/pull/29/commits/7125374c0e755734d7e55e86ceacb3829d20d5b5

File: src/functions/wceStyles.ts (hunk 1 above)
Comment (bananarama92) [reply src/functions/wceStyles.ts]: Oh, neat! Wasn't 100% certain of BC symbols could already be accessed at this point, but this does the trick nicely in that case.

Applied as of https://github.com/KittenApps/WCE/pull/29/commits/7125374c0e755734d7e55e86ceacb3829d20d5b5

Comment (KittenApps) [reply src/functions/wceStyles.ts]: `GameVersion ` is available very early, even before you login, which was important here

Thanks for the fix.

**Pull request merged**